lib/health.js

```diff
@@ -65,36 +70,102 @@ class HealthCheck {
     };
   }
 
-  addCheck (componentName, metricName, checkerPromise) {
+  /**
+   * Add a new health checking routine - for a metric, on a component
+   *
+   * @param {string} componentName - name of the component whose health is being
+   * reported
+   * @param {string} metricName - name of the metric on a component the value of
+   * which is being reported
+   * @param {AsyncFunction or Promise} checkExecutor - an async function to call
```

Comment:
Both the documentation and the implementation still seem to expect that a promise, rather than a function, could be passed as a `checkExecutor`.   I think you could remove the "or Promise" language here, and the else-if clause on L94, and the "promise" language from L98, and I'd be okay with it.

Suggested L80:
```javascript
* @param {Function} checkExecutor - a function to call in an async context to return a promise of values
```

Remove L93-96.

Suggested L98:
```javascript
throw new Error(`Callback for ${key} must be a function`);
```

Reply:
Fix in: 0789744